hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterStateManager.java

```diff
@@ -119,33 +131,62 @@ void setClusterState(ClusterState newState, boolean isTransient) {
         }
     }
 
+    private void setClusterStateAndVersion(ClusterState newState, Version newVersion, boolean isTransient) {
+        this.state = newState;
+        this.version = newVersion;
+        stateLockRef.set(ClusterStateLock.NOT_LOCKED);
+        changeNodeState(newState);
+        node.getNodeExtension().onClusterStateChange(newState, isTransient);
+        node.getNodeExtension().onClusterVersionChange(newVersion);
+    }
+
     private void doSetClusterState(ClusterState newState, boolean isTransient) {
         this.state = newState;
         stateLockRef.set(ClusterStateLock.NOT_LOCKED);
         changeNodeState(newState);
         node.getNodeExtension().onClusterStateChange(newState, isTransient);
     }
 
+    private void doSetClusterVersion(Version newVersion) {
```

Comment:
@pveentjer I added a cluster version change listener in `SystemLogPlugin`. Test to verify output will be on the EE-side PR.